src/test/java/mate/academy/bookstoreapp/controller/BookControllerTest.java

```diff
@@ -0,0 +1,298 @@
+
+        MvcResult result = mockMvc.perform(get("/books/search")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(searchParamsJson))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        String content = result.getResponse().getContentAsString();
+        LOGGER.info("Response content: " + content);
+
+        List<BookDto> books = objectMapper.readValue(content, new TypeReference<>() {});
+
+        assertNotNull(books);
+        assertFalse(books.isEmpty());
+        assertTrue(books.stream().anyMatch(book -> book.title().contains("Sherlock")));
+    }
+
+    @Sql(scripts = "classpath:databasescripts/books/add-books-and-categories"
+            + "-to-books-and-categories-tables.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:databasescripts/books/delete-two-books-and-categories"
+            + "-from-books-and-categories-tables.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @Test
+    @DisplayName("Create book - Admin User")
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    public void createBook_AdminUser_ReturnsCreatedBook() throws Exception {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto(
+                "Another Adventures",
+                "Another Author",
+                "978-3-16-148410-0",
+                BigDecimal.valueOf(19.99),
+                "It's absolutely another story",
+                "cover.jpeg",
+                List.of(1L)
+        );
+
+        String requestJson = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(post("/books")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(requestJson))
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        String content = result.getResponse().getContentAsString();
+        LOGGER.info("Response content: " + content);
+        BookDto createdBook = objectMapper.readValue(content, BookDto.class);
+
+        assertNotNull(createdBook);
+        assertEquals("Another Adventures", createdBook.title());
+        assertEquals("Another Author", createdBook.author());
+        assertEquals("978-3-16-148410-0", createdBook.isbn());
+        assertEquals(BigDecimal.valueOf(19.99), createdBook.price());
+    }
+
+    @Test
+    @DisplayName("Create book - Non-Admin User")
+    @WithMockUser(username = "user", roles = {"USER"})
+    public void createBook_NonAdminUser_ReturnsForbidden() throws Exception {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto(
+                "Another Adventures",
+                "Another Author",
+                "978-3-16-148410-0",
+                BigDecimal.valueOf(19.99),
+                "It's absolutely another story",
+                "cover.jpeg",
+                List.of(1L)

```

Comment:
I see duplication of code, let`s create private method with building this entity

Reply:
and feel free to use constants for values